### ISSUE TITLE
open /proc/10782/fdinfo/1023: too many open files

### ISSUE THREAD

**Issue by borkmann** [cilium/cilium#632]
Running 11-getting-started in a loop for ~30min on my laptop:
```
May 02 11:56:49 linux-3-1.home cilium-agent[10782]: INFO Regenerated program of endpoint 42380
May 02 11:56:50 linux-3-1.home cilium-agent[10782]: INFO Regenerated program of endpoint 35132
May 02 11:56:50 linux-3-1.home cilium-agent[10782]: INFO Regenerated program of endpoint 50640
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: WARN Unable to open CT map's fdinfo /sys/fs/bpf/tc/globals/cilium_ct4_35132: open /proc/10782/fdinfo/1023: too many open files
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: panic: runtime error: invalid memory address or nil pointer dereference
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x40e52e]
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: goroutine 57 [running]:
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: panic(0x199a360, 0xc42000e040)
May 02 11:57:15 linux-3-1.home cilium-agent[10782]:         /usr/lib/golang/src/runtime/panic.go:500 +0x1a1
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: main.runGC(0xc4223ff7c0, 0x1bcc1ba, 0xb, 0x1)
May 02 11:57:15 linux-3-1.home cilium-agent[10782]:         /usr/lib/golang/src/github.com/cilium/cilium/daemon/ct.go:47 +0x39e
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: main.(*Daemon).EnableConntrackGC.func1(0xc421af8000)
May 02 11:57:15 linux-3-1.home cilium-agent[10782]:         /usr/lib/golang/src/github.com/cilium/cilium/daemon/ct.go:82 +0x1d4
May 02 11:57:15 linux-3-1.home cilium-agent[10782]: created by main.(*Daemon).EnableConntrackGC
May 02 11:57:15 linux-3-1.home cilium-agent[10782]:         /usr/lib/golang/src/github.com/cilium/cilium/daemon/ct.go:89 +0x3f
May 02 11:57:15 linux-3-1.home systemd[1]: cilium.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
May 02 11:57:15 linux-3-1.home systemd[1]: cilium.service: Unit entered failed state.
May 02 11:57:15 linux-3-1.home systemd[1]: cilium.service: Failed with result 'exit-code'.
May 02 11:57:16 linux-3-1.home systemd[1]: cilium.service: Service hold-off time over, scheduling restart.
May 02 11:57:16 linux-3-1.home systemd[1]: Stopped cilium.
May 02 11:57:16 linux-3-1.home systemd[1]: Started cilium.
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO      _ _ _
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO  ___|_| |_|_ _ _____
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO |  _| | | | | |     |
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO |___|_|_|_|___|_|_|_|
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO Cilium 0.8.90 1ebafea Mon, 1 May 2017 13:10:12 +0200 go version go1.7.5 linux/amd64
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO clang and kernel versions: OK!
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO linking environment: OK!
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO BPF Verifier: OK!
May 02 11:57:16 linux-3-1.home cilium-agent[28843]: INFO Valid label prefix configuration:

```
Maybe also related to the error in #631.

**Comment by borkmann** [comment]
The ef6831a2b644e ("daemon, ct: fix fd leakage on lru") likely fixed the seen issue with regards to too many open fds. I haven't seen it since at least.

The Go panic ("invalid memory address or nil pointer dereference") should still be fixed in the code if possible from cilium side.

**Comment by aanm** [comment]
What I find odd is the go panic takes place in a log message:
`log.Warningf("Unable to open CT map's fdinfo %s: %s\n", file, err)`
`file` is a string and `err` is not nil

**Comment by tgraf** [comment]
> What I find odd is the go panic takes place in a log message:
> log.Warningf("Unable to open CT map's fdinfo %s: %s\n", file, err)
> file is a string and err is not nil

I think it's a golang bug

**Comment by borkmann** [comment]
Not seeing anymore since ef6831a ("daemon, ct: fix fd leakage on lru"), same as in #631.
